app/api/discount_codes.py

```diff
@@ -0,0 +1,44 @@
+from app.api.helpers.permissions import jwt_required
+from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
+from marshmallow_jsonapi.flask import Schema, Relationship
+from marshmallow_jsonapi import fields
+from app.models import db
+from app.models.discount_code import DiscountCode
+from app.models.event_invoice import EventInvoice
+
+
+class DiscountCodeSchema(Schema):
+    """
+    API Schema for discount_code Model
+    """
+    class Meta:
+        type_ = 'discount_code'
+        self_view = 'v1.discount_code_detail'
+        self_view_kwargs = {'id': '<id>'}
+        self_view_many = 'v1.discount_code_list'
+
+    id = fields.Integer()
+    code = fields.Str()
+    discount_url = fields.Str()
+    value = fields.Float()
+    type = fields.Str()
+    is_active = fields.Boolean()
+
+    event_invoice = Relationship(attribute='event_invoice',
```

Comment:
@magdalenesuo Reconsidering this, I think that this relationship should not exist as in the discount_code model, `used_for` fields exists as a string which does not arise from a relationship. The inverse one is required though : discount_code in `api/event-invoice.py`